src/Core/Lock/LockTrait.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core\Lock;
+
+/**
+ * Utility trait for locks.
+ */
+trait LockTrait
+{
+    /**
+     * Acquires a lock that will block until released.
+     *
+     * @return bool
+     * @throws \RuntimeException
+     */
+    abstract public function acquire();
```

Comment:
`synchronize` calls `acquire` and `release`, so I wanted to be sure that if someone included the trait those methods were always implemented. Another option would be an abstract class implementing the interface, but I preferred the trait here - WDYT?